src/module/actor/sheet/loot/loot-npcs-popup.ts

```diff
@@ -34,14 +36,51 @@ export class LootNPCsPopup extends FormApplication<ActorPF2e> {
             }
             const currentSource = token.actor;
             if (selectionData[i] && currentSource) {
-                const currentSourceItemData = currentSource.inventory.map((item) => item.toObject());
-                itemData.push(...currentSourceItemData);
-                const idsToDelete = currentSourceItemData.map((item) => item._id);
+                for (const item of currentSource.inventory) {
+                    const stackableItem = currentSource.findStackableItem(this.object, item);
+                    if (stackableItem) {
+                        const currentQuantity = itemUpdates.get(stackableItem.id) ?? stackableItem.quantity;
+                        itemUpdates.set(stackableItem.id, currentQuantity + item.quantity);
+                        continue;
+                    }
+                    newItems.push(item);
+                }
+                const idsToDelete = currentSource.inventory.map((item) => item.id);
                 currentSource.deleteEmbeddedDocuments("Item", idsToDelete);
             }
         }
-        if (itemData.length > 0) {
-            await this.object.createEmbeddedDocuments("Item", itemData);
+        // Create new items in the loot actor
+        if (newItems.length > 0) {
+            const stacked: PhysicalItemPF2e[] = [];
+            for (const item of newItems) {
```

Comment:
This entire for looks like it could be a reduce. With a comment saying to merge stackable items within the same incoming batch.

Reply:
will do